### DISCUSSION TITLE
Not redirecting to Idp when logging in using OIDC

### DISCUSSION THREAD

**Discussion by InoryS** [mastodon/mastodon#24163] [Q&A]
I'm trying to use OIDC with keycloak as Idp

I configured mastodon with following settings:

```
#OIDC
OIDC_ENABLED=true
OIDC_DISPLAY_NAME='name'
OIDC_ISSUER=https://keycloak/realms/master
OIDC_DISCOVERY=true
OIDC_SCOPE=openid,profile,email
OIDC_UID_FIELD=preferred_username
OIDC_CLIENT_ID=
OIDC_CLIENT_SECRET=
OIDC_REDIRECT_URI=https://mastodon/auth/auth/openid_connect/callback
OIDC_SECURITY_ASSUME_EMAIL_IS_VERIFIED=true
# uncomment to only use OIDC for login / registration buttons
OMNIAUTH_ONLY=false
```

But when I click login with 'name' nothing happens, I can see in the browser console that https://mastodon/auth/auth/openid_connect is being requested, this request never succeeds.

If I go directly access to https://mastodon/auth/auth/openid_connect I would get `"Not found. Authentication passthru."` or `"504 We're sorry, but something went wrong on our end."`

I can confirm that https://keycloak/realms/master/.well-known/openid-configuration is configured correctly.

The log of mastodon-web 
```
method=GET path=/auth/auth/openid_connect format=html controller=Auth::OmniauthCallbacksController action=passthru status=404 duration=2.86 view=1.41 db=0.00
method=GET path=/auth/auth/openid_connect format=html controller=Auth::OmniauthCallbacksController action=passthru status=404 duration=2.81 view=1.34 db=0.00
```

Did I do something wrong, please help me, thanks!

**Comment by mscurtescu** [comment]
Don't you have to fill in `OIDC_CLIENT_ID` and `OIDC_CLIENT_SECRET`? Discovery will provide endpoints and capabilities, but you probably still need a registered client.

**Reply by InoryS** [reply]
Thanks for the reply, it fixed itself after I upgraded to 4.1.1, don't know why.
And yes, I filled in all of this information correctly, I just removed it from the example for security reasons.

**Comment by InoryS** [comment]
It fixed itself after I upgraded to 4.1.1, don't know why.
